### PR TITLE
resolve #1601 bgem3 nlp model integration

### LINKED ISSUE #1601: [NLP-model] Integrate bgem3 model

### PR DESCRIPTION
**Pull request by asus007**

### CODE REVIEW

**Comment by Fabilin** [comment]
If I understand correctly, this module adds a new NLP provider that can delegate to any language model hosted on AWS Sagemaker. Could we maybe rename it to `tock-nlp-model-sagemaker` and rename the classes accordingly to signal the fact that it can work with any model?

File: nlp/front/ioc/pom.xml [hunk 1]
```diff
@@ -41,6 +41,11 @@
             <groupId>ai.tock</groupId>
             <artifactId>tock-nlp-model-opennlp</artifactId>
         </dependency>
+        <dependency>
+            <groupId>ai.tock</groupId>
+            <artifactId>tock-nlp-model-bgem3</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```
Comment (vsct-jburet) [inline nlp/front/ioc/pom.xml:48]: Don't add bgem3 by default please :)

File: nlp/model/sagemaker/src/main/kotlin/SagemakerModelConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.tock.nlp.sagemaker
+
+internal data class SagemakerModelConfiguration(
+    // Default model name
+    val modelFileName: String = "evoyageurs:bgem3-fr-bgem3"
```
[the remark this answers is not in the document]
Comment (vsct-jburet) [reply nlp/model/sagemaker/src/main/kotlin/SagemakerModelConfiguration.kt:21]: val modelFileName: String = property ("tockt_sagemaker_aws_model_file_name", "default")

File: nlp/model/sagemaker/src/main/kotlin/SagemakerIntentClassifier.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+internal class SagemakerIntentClassifier(private val conf: SagemakerModelConfiguration) : IntentClassifier {
+
+    override fun classifyIntent(context: IntentContext, text: String, tokens: Array<String>): IntentClassification {
+        return SagemakerClientProvider.getClient(
+            SagemakerAwsClientProperties(
+                Region.of(property("tock_sagemaker_aws_region_name", "eu-west-3")),
+                property("tock_sagemaker_aws_intent_endpoint_name", "bge-m3-model-intent--v0"),
+                property("tock_sagemaker_aws_content_type", "application/json"),
+                property("tock_sagemaker_aws_profile_name", "sa-voyageurs-dev"),
```
Comment (vsct-jburet) [inline nlp/model/sagemaker/src/main/kotlin/SagemakerIntentClassifier.kt:34]: do not use the default of your company please :)

File: nlp/model/sagemaker/src/main/kotlin/SagemakerEntityClassifier.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        context: EntityCallContext,
+        text: String,
+        tokens: Array<String>
+    ): List<EntityRecognition> {
+        SagemakerClientProvider.getClient(
+            SagemakerAwsClientProperties(
+                Region.of(property("tock_sagemaker_aws_region_name", "eu-west-3")),
+                property("tock_sagemaker_aws_entities_endpoint_name", "bge-m3-model-entities"),
+                property("tock_sagemaker_aws_content_type", "application/json"),
+                property("tock_sagemaker_aws_profile_name", "sa-voyageurs-dev"),
```
Comment (vsct-jburet) [inline nlp/model/sagemaker/src/main/kotlin/SagemakerEntityClassifier.kt:41]: do not use the default of your company please :)

**Review by vsct-jburet — APPROVED** [review]

**Pull request merged**